lib/logstash/outputs/elasticsearch.rb

```diff
@@ -191,6 +196,28 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # have become stale (half-closed) while kept inactive in the pool.'
   # See https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html#setValidateAfterInactivity(int)[these docs for more info]
   config :validate_after_inactivity, :validate => :number, :default => 10000
+  
+  # This is the maximum size HTTP request this plugin will attempt to send.
+  #
+  # This plugin will try to send appropriately sized requests, using multiple
+  # requests per batch if the events are very large, but if a single event
+  # is larger than this value sending even that single request will break. 
+  #
+  # You should only consider changing this setting if:
+  # 1. You want to set it to a lower value to prevent large events being stored in ES.
+  # 2. You have increased Elasticsearch's `http.max_content_length` setting.
+  # 
+  # We check this size after serializing to see whether we should even try to send it
+  # If the event is larger than this we log an error and drop the event.
+  # 
+  # The default is set to work with Elasticsearch's default maximum request size of 100MiB
+  # Note that we've set it by default to 1MiB lower as a fudge factor for any protocol overhead
+  # or other things that could account for a slight difference in the HTTTP request, as our size
+  # check only happens on the request body.
+  #
+  # This is specified as a number now, but we should move it to the new 'bytes' type
```

Comment:
"This is specified ..." will show up in the docs. I recommend not putting this comment here.